File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/WritableMap.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/** Interface for a mutable map. Used to pass arguments from Java to JS. */
public interface WritableMap extends ReadableMap {

  void putNull(@NonNull String key);

  void putBoolean(@NonNull String key, boolean value);

  void putDouble(@NonNull String key, double value);

  void putInt(@NonNull String key, int value);

  void putString(@NonNull String key, @Nullable String value);

  void putArray(@NonNull String key, @Nullable ReadableArray value);

  void putMap(@NonNull String key, @Nullable ReadableMap value);

  void merge(@NonNull ReadableMap source);

  WritableMap copy();
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/WritableMap.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge

/** Interface for a mutable map. Used to pass arguments from Kotlin to JS. */
public interface WritableMap : ReadableMap {
  public fun copy(): WritableMap

  public fun merge(source: ReadableMap)

  public fun putArray(key: String, value: ReadableArray?)

  public fun putBoolean(key: String, value: Boolean)

  public fun putDouble(key: String, value: Double)

  public fun putInt(key: String, value: Int)

  public fun putLong(key: String, value: Long)

  public fun putMap(key: String, value: ReadableMap?)

  public fun putNull(key: String)

  public fun putString(key: String, value: String?)
}
